v2/pkg/application/systray.go

```diff
@@ -0,0 +1,151 @@
+package application
+
+import (
+	"github.com/wailsapp/wails/v2/internal/platform"
+	"github.com/wailsapp/wails/v2/pkg/menu"
+	"github.com/wailsapp/wails/v2/pkg/options"
+)
+
+// SystemTray defines a system tray!
+type SystemTray struct {
+	title              string
+	hidden             bool
+	lightModeIcon      *options.SystemTrayIcon
+	darkModeIcon       *options.SystemTrayIcon
+	tooltip            string
+	startHidden        bool
+	menu               *menu.Menu
+	onLeftClick        func()
+	onRightClick       func()
+	onLeftDoubleClick  func()
+	onRightDoubleClick func()
+	onMenuClose        func()
+	onMenuOpen         func()
+
+	// The platform specific implementation
+	impl platform.SysTray
+}
+
+func newSystemTray(options *options.SystemTray) *SystemTray {
+	return &SystemTray{
+		title:              options.Title,
+		lightModeIcon:      options.LightModeIcon,
+		darkModeIcon:       options.DarkModeIcon,
+		tooltip:            options.Tooltip,
+		startHidden:        options.StartHidden,
+		menu:               options.Menu,
+		onLeftClick:        options.OnLeftClick,
+		onRightClick:       options.OnRightClick,
+		onLeftDoubleClick:  options.OnLeftDoubleClick,
+		onRightDoubleClick: options.OnRightDoubleClick,
+		onMenuOpen:         options.OnMenuOpen,
+		onMenuClose:        options.OnMenuClose,
+	}
+}
+
+func (t *SystemTray) run() {
```

Comment:
Don't we have to lock down the OS thread here? Since Windows will send the messages of the systray to the message loop of the OS thread that created the window. So we would have to make sure all of the creation of the SysTray will be done on the same OS thread as t.impl.Run() is run.

This is OS specific for windows, so we might need to put that into platform specific code behind a build tag.